plugins/nanospell/plugin.js

```diff
@@ -854,7 +854,7 @@
 					if (word) words.push(word);
 				}
 
-				return words.join(" ");
+				return words;
```

Comment:
basically, changed the string joining to happen outside of this function, so we don't need to do weird things like count spaces to find out how many words we have.  There should probably be a follow up refactor to just use arrays or Sets (except IE11 support for Sets is bad) everywhere - some of this is leftover code still.